src/react/databrowser/objects/__tests__/Metadata.test.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { reduxMount } from '../../../utils/test';
+
+describe('Metadata', () => {
+    const putObjectMetadataMock = jest.spyOn(s3objects, 'putObjectMetadata').mockReturnValue({ type: '' });
+    const optionLabels = ['cache-control', 'content-disposition', 'content-encoding', 'content-type', 'website-redirect-location', 'x-amz-meta', 'content-language', 'expires'];
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('Metadata should render', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find(Metadata).isEmptyRender()).toBe(false);
+    });
+
+    it('should render by default a SelectBox with empty values in each input when there are no key/value present', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
+        // check if item row values are rendered
+        expect(component.find('input[name="mdKeyType"]').prop('value')).toBe('');
+        expect(component.find('input#metadata-input-value').prop('value')).toBe('');
+    });
+
+    it('should add new key/value metadata and should trigger function if save button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
+
+        // open dropdown menu
+        component.find('input').at(0).simulate('change');
+
+        // check all options are present
+        const optionsComponents = component.find('Option');
+        optionsComponents.forEach((optionComponent, index) => {
+            expect(optionComponent.find('div.sc-select-option-label').text()).toBe(optionLabels[index]);
+        });
+
+        // select a key
+        optionsComponents.at(0).simulate('click');
+
+        // check dropdown menu disappear
+        expect(component.find('Option')).toHaveLength(0);
+
+        // add a value to the key
+        component.find('input#metadata-input-value').simulate('change', { target: { value: 'value1' } });
+
+        // click on add button
+        component.find(AddButton).simulate('click');
+
+        // check if new item row added
+        expect(component.find(Item)).toHaveLength(2);
+
+        // check if function when pressing save button is triggered
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
+        component.find('Button#metadata-button-save').simulate('click');
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('remove button and add button should be disabled', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
```

Comment:
done

Reply:
I don't see the change for this test.
Also, we can add className `delbtn` to `SubButton`'s `Button` component and find `find('.delbtn')` instead of using 'children'.  It will be less mistake-prone if we happen to change the `AddButton` and `SubButton` component.
Same remark for `addbtn` to `AddButton`'s `CustomButton` component.

Reply:
I don't see the change in this test.
My point was to use `firstItem` instead of `component` to make sure we are in the first Item although there is only one here.
Same thing for the following assertions.